app/controllers/hmis_api/graphql_controller.rb

```diff
@@ -0,0 +1,55 @@
+module HmisApi
+  class GraphqlController < HmisApi::BaseController
+    # If accessing from outside this domain, nullify the session
+    # This allows for outside API access while preventing CSRF attacks,
+    # but you'll have to authenticate your user separately
+    # protect_from_forgery with: :null_session
+
+    def execute
+      variables = prepare_variables(params[:variables])
+      query = params[:query]
+      operation_name = params[:operationName]
+      context = {
+        current_user: current_hmis_api_user,
+      }
+      case params[:schema]
+      when :hmis
+        result = HmisSchema.execute(query, variables: variables, context: context, operation_name: operation_name)
```

Comment:
I assume HmisSchema.execute will fail gracefully if current_user is nil, but do we want to just return `{}` or unauthorized or similar at the top of the method if `current_hmis_api_user` is blank?  I believe we should never return data if we don't have a user.

Reply:
The superclass `HmisApi::BaseController` defines `authenticate_user!` to be `authenticate_hmis_api_user!`, so I don't think it can be blank?

Reply:
That's right, it will return 401 if there is no user